proto/google/events/firebase/v1/data.proto

```diff
@@ -17,22 +17,327 @@ syntax = "proto3";
+// The data within all Firebase Real Time Database reference events.
+message ReferenceEventData {
+  // The original data for the reference.
   google.protobuf.Value data = 1;
-  // The change in the document data.
+  // The change in the reference data.
   google.protobuf.Value delta = 2;
+}
+
+// The data within all Firebase Remote Config events.
+message RemoteConfigEventData {
+  // The version number of the version's corresponding Remote Config template.
+  int64 version_number = 1;
+  
+  // When the Remote Config template was written to the Remote Config server.
+  google.protobuf.Timestamp update_time = 2;
+  
+  //  Aggregation of all metadata fields about the account that performed the update.
+  RemoteConfigUser update_user = 3;
+  
+  // The user-provided description of the corresponding Remote Config template.
+  string description = 4;
+  
+  // Where the update action originated.
+  RemoteConfigUpdateOrigin update_origin = 5;
+  
+  // What type of update was made.
+  RemoteConfigUpdateType update_type = 6;
+  
+  // Only present if this version is the result of a rollback, and will be the
+  // version number of the Remote Config template that was rolled-back to.
+  int64 rollback_source = 7;
+}
+
+// All the fields associated with the person/service account
+// that wrote a Remote Config template.
+message RemoteConfigUser {
+  // Display name.
+  string name = 1;
+  
+  // Email address.
+  string email = 2;
+  
+  // Image URL.
+  string imageUrl = 3;
+}
+
+// What type of update was associated with the Remote Config template version.
+enum RemoteConfigUpdateOrigin {
+  // Catch-all for unrecognized values.
+  REMOTE_CONFIG_UPDATE_ORIGIN_UNSPECIFIED = 0;
+  
+  // The update came from the Firebase UI.
+  CONSOLE = 1;
+  
+  // The update came from the Remote Config REST API.
+  REST_API = 2;
+
+  // The update came from the Firebase Admin Node SDK.
+  ADMIN_SDK_NODE = 3;
+}
+
+// Where the Remote Config update action originated.
+enum RemoteConfigUpdateType {
+  // Catch-all for unrecognized enum values.
+  REMOTE_CONFIG_UPDATE_TYPE_UNSPECIFIED = 0;
+  
+  // A regular incremental update.
+  INCREMENTAL_UPDATE = 1;
+  
+  // A forced update.
+  // The ETag was specified as "*" in an UpdateRemoteConfigRequest
+  // request or the "Force Update" button was pressed on the console.
+  FORCED_UPDATE = 2;
+  
+  // A rollback to a previous Remote Config template.
+  ROLLBACK = 3;
+}
+
+message AuthEventData {
+  // The user identifier in the Firebase app.
+  string uid = 1;
+
+  // The user's primary email, if set.
+  string email = 2;
+
+  // Additional metadata about the user.
+  UserMetadata metadata = 7;
+
+  // User's info at the providers
+  repeated UserInfo provider_data = 8;
+}
+
+// Additional metadata about the user.
+message UserMetadata {
+  // The date the user was created.
+  google.protobuf.Timestamp created_at = 1;
+
+  // The date the user last signed in.
+  google.protobuf.Timestamp last_signed_in_at = 2;
+}
```

Comment:
As per [AIP-142](https://google.aip.dev/142) these would be different names, but it seems like it's probably a bad idea to change the signature in the event payload when the REST API returns the incorrect response.